Map various things

copygirl opened the pull request:
See commits for details

RX14 commented:
Filthy mergepull commits. You should run `git config branch.16w33a.rebase true` to get rid of them when using `git pull`. Also should run `git config --global branch.autosetuprebase always` to do this automatically on all new branches/clones in all future repos.

RX14 commented:
Actually `git config --global --bool pull.rebase true` is probably the better config to change.

copygirl committed:
Map ServerRunnable

copygirl committed:
Map ServerConfigList and related

Renamed from ProfileList
Mapped ServerConfigEntry
Renamed list and entry classes

copygirl committed:
Map / update ServerConfigurationManagers

copygirl committed:
Map some Potion(Effect) stuff

copygirl committed:
Map a few minor fields

copygirl committed:
Map DamageSource

Renamed EntityProjectile => EntityArrowBase
Mapped Entity.damage

copygirl commented:
Fixed my rebase screwup that would've probably rewritten this repo's history (which the conflict checker seems to be fine with?).

asiekierka commented on mappings/net/minecraft/entity/projectile/EntityArrowBase.mapping:
```diff
@@ -1,4 +1,4 @@
-CLASS none/aan net/minecraft/entity/projectile/EntityProjectile
+CLASS none/aan net/minecraft/entity/projectile/EntityArrowBase
```
Sure about that?

copygirl replied:
There's projectile classes (fireballs, snowballs, enderpearls, ...) that don't extend this class. `EntityArrowBase` is the base class for normal and specter arrows, and that's it.

asiekierka commented on mappings/net/minecraft/potion/Potion.mapping:
```diff
@@ -1,7 +1,7 @@
 	FIELD a REGISTRY Lnone/cs;
-	FIELD b id Lnone/kp;
-	FIELD c maxRawId I
+	FIELD b WATER_TEX Lnone/kp;
```
This is a location, not the texture. WATER_TEX_LOC

copygirl replied:
I remember seeing a lot of texture locations defined as `BG_TEX` and similar, as such I thought that was the way to do it. I'd say the type already shows that it's a location.

Let's vote on it? :+1: for `WATER_TEX_LOC`, :-1: for `WATER_TEX`?

asiekierka replied:
Eh, I suppose WATER_TEX is fine. It's not a big deal.

asiekierka commented on mappings/net/minecraft/potion/PotionEffect.mapping:
```diff
@@ -24,8 +25,26 @@ CLASS none/ry net/minecraft/potion/PotionEffect
 	METHOD <init> (Lnone/ry;)V
 		ARG 0 other
 	METHOD a getEffectType ()Lnone/rx;
+	METHOD a toNbt (Lnone/dt;)Lnone/dt;
```
No! serializeTag/deserializeTag

asiekierka commented on mappings/net/minecraft/potion/PotionEffect.mapping:
```diff
@@ -24,8 +25,26 @@ CLASS none/ry net/minecraft/potion/PotionEffect
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 0 other
 	METHOD f getTranslationKey ()Ljava/lang/String;
+	METHOD g getPermanent ()Z
```
isPermanent/setPermanent for booleans...

asiekierka commented:
Other than those, LGTM! Good work.

copygirl commented:
Getting a hang of rebase. Fixed `isPermanent` and `de/serializeTag`.

darkhax commented:
Lets go to mars ^^

asiekierka commented:
Yay!
